README.md

```diff
@@ -4,175 +4,225 @@
+of the token, using an existing cipher of type `FakeCrypto`[^cipher]:
 
-Following up from the previous example, let's assume we now want to create a signed access token containing the
-existing `key`, as well as claims about the audience and issuer of the token, using an existing `cipher`[^cipher]:
 ```rust
-use dcaf::{ToCborMap, sign_access_token, verify_access_token};
-use coset::cwt::ClaimsSetBuilder;
-use coset::Header;
-use coset::iana::CwtClaimName;
 
+# [derive(Clone)]
+
+let rng = FakeRng;
+let key = FakeKey { key: [1, 2, 3, 4, 5], kid: [0xDC, 0xAF]};
+let cose_key: CoseKey = key.to_cose_key();
 let claims = ClaimsSetBuilder::new()
-   .audience("valve242".to_string())
-   .claim(CwtClaimName::Cnf, key.to_ciborium_value())
-   .claim(CwtClaimName::Scope, Value::Text("read".to_string()))
-   .build();
-let token = sign_access_token(claims, &mut cipher, None, None, None)?;
-assert!(verify_access_token(&token, &mut cipher, None).is_ok());
+.audience(String::from("coaps://rs.example.com"))
+.issuer(String::from("coaps://as.example.com"))
+.claim(CwtClaimName::Cnf, cose_key.to_cbor_value() ? )
+.build();
+let token = sign_access_token::<FakeCrypto, FakeRng>( & key, claims, None, None, None, rng) ?;
+assert!(verify_access_token::<FakeCrypto>(&key, &token, None).is_ok());
 ```
 
 [^cipher]: Note that we are deliberately omitting details about the implementation of the
-`cipher` here, since such implementations won't be in scope of this crate.
-
-## Provided Data Models
-
-### Token Endpoint
+`cipher` here, since such implementations won't be in the scope of this crate.
 
-The most commonly used models will probably be the token endpoint's `AccessTokenRequest` and
-`AccessTokenResponse` described in
-[section 5.8 of the ACE-OAuth draft](https://www.ietf.org/archive/id/draft-ietf-ace-oauth-authz-46.html#section-5.8). In
-case of an error, an `ErrorResponse` should be used.
+# Provided Data Models
 
-After an initial Unauthorized Resource Request Message, an `AuthServerRequestCreationHint` can be used to provide
-additional information to the client, as described in
-[section 5.3 of the ACE-OAuth draft](https://www.ietf.org/archive/id/draft-ietf-ace-oauth-authz-46.html#section-5.3).
-
-### Common Data Types
-Some types used across multiple scenarios include:
+## Token Endpoint
 
-- `Scope` (as described
-  in [section 5.8.1 of the ACE-OAuth draft](https://www.ietf.org/archive/id/draft-ietf-ace-oauth-authz-46.html#section-5.8.1))
-  , either as a `TextEncodedScope`, a `BinaryEncodedScope` or an `AifEncodedScope`.
-- `ProofOfPossessionKey` as specified
-  in [section 3.1 of RFC 8747](https://datatracker.ietf.org/doc/html/rfc8747#section-3.1). For example, this will be
-  used in the access token's `cnf` claim.
-- While not really a data type, various constants representing values used in ACE-OAuth are provided in the `constants`
-  module.
+The most commonly used models will probably be the token endpoint's
```

Comment:
```suggestion
The most commonly used models will probably be the token endpoints
```

Reply:
Possessive `token endpoint's` should be correct here, as I'm talking about the token endpoint's `AccessTokenRequest` and `AccessTokenResponse` model.

Reply:
Oh, I see! Thank you for the clarification :)